pkg/model/core/grpcroute.go

```diff
@@ -0,0 +1,302 @@
+	grpcRoute := &gateway_api_v1alpha2.GRPCRoute{}
+	err := client.Get(ctx, routeNamespacedName, grpcRoute)
+	if err != nil {
+		return nil, err
+	}
+	return NewGRPCRoute(*grpcRoute), nil
+}
+
+func ListGRPCRoutes(client client.Client, context context.Context) []Route {
+	routeList := &gateway_api_v1alpha2.GRPCRouteList{}
+	client.List(context, routeList)
+	var routes []Route
+	for _, route := range routeList.Items {
+		routes = append(routes, NewGRPCRoute(route))
+	}
+	return routes
+}
+
+func (r *GRPCRoute) Spec() RouteSpec {
+	return &GRPCRouteSpec{r.r.Spec}
+}
+
+func (r *GRPCRoute) Status() RouteStatus {
+	return &GRPCRouteStatus{&r.r.Status}
+}
+
+func (r *GRPCRoute) Name() string {
+	return r.r.Name
+}
+
+func (r *GRPCRoute) Namespace() string {
+	return r.r.Namespace
+}
+
+func (r *GRPCRoute) DeletionTimestamp() *v1.Time {
+	return r.r.DeletionTimestamp
+}
+
+func (r *GRPCRoute) DeepCopy() Route {
+	return &GRPCRoute{r: *r.r.DeepCopy()}
+}
+
+func (r *GRPCRoute) K8sObject() client.Object {
+	return &r.r
+}
+
+func (r *GRPCRoute) Inner() *gateway_api_v1alpha2.GRPCRoute {
+	return &r.r
+}
+
+type GRPCRouteSpec struct {
+	s gateway_api_v1alpha2.GRPCRouteSpec
+}
+
+func (s *GRPCRouteSpec) ParentRefs() []gateway_api_v1beta1.ParentReference {
+	return s.s.ParentRefs
+}
+
+func (s *GRPCRouteSpec) Hostnames() []gateway_api_v1beta1.Hostname {
+	return s.s.Hostnames
+}
+
+func (s *GRPCRouteSpec) Rules() []RouteRule {
+	var rules []RouteRule
+	for _, rule := range s.s.Rules {
+		rules = append(rules, &GRPCRouteRule{rule})
+	}
+	return rules
+}
+
+func (s *GRPCRouteSpec) Equals(routeSpec RouteSpec) bool {
+	_, ok := routeSpec.(*GRPCRouteSpec)
+	if !ok {
+		return false
+	}
+
+	if !reflect.DeepEqual(s.ParentRefs(), routeSpec.ParentRefs()) {
+		return false
+	}
+
+	if !reflect.DeepEqual(s.Hostnames(), routeSpec.Hostnames()) {
+		return false
+	}
+
+	if len(s.Rules()) != len(routeSpec.Rules()) {
+		return false
+	}
+
+	for i, rule := range s.Rules() {
+		otherRule := routeSpec.Rules()[i]
+		if !rule.Equals(otherRule) {
+			return false
+		}
+	}
+
+	return true
+}
+
+type GRPCRouteStatus struct {
+	s *gateway_api_v1alpha2.GRPCRouteStatus
+}
+
+func (s *GRPCRouteStatus) Parents() []gateway_api_v1beta1.RouteParentStatus {
+	return s.s.Parents
+}
+
+func (s *GRPCRouteStatus) SetParents(parents []gateway_api_v1beta1.RouteParentStatus) {
+	s.s.Parents = parents
+}
+
+type GRPCRouteRule struct {
+	r gateway_api_v1alpha2.GRPCRouteRule
+}
+
+func (r *GRPCRouteRule) BackendRefs() []BackendRef {
+	var backendRefs []BackendRef
+	for _, backendRef := range r.r.BackendRefs {
+		backendRefs = append(backendRefs, &GRPCBackendRef{backendRef})
+	}
+	return backendRefs
+}
+
+func (r *GRPCRouteRule) Matches() []RouteMatch {
+	var routeMatches []RouteMatch
+	for _, routeMatch := range r.r.Matches {
+		routeMatches = append(routeMatches, &GRPCRouteMatch{routeMatch})
+	}
+	return routeMatches
+}
+
+func (r *GRPCRouteRule) Equals(routeRule RouteRule) bool {
```

Comment:
please add unit test, same for other methods with conditional logic, for example other "Equal" methods

Reply:
Added a suite of unit tests for all route abstraction Equals methods